executor/point_get.go

```diff
@@ -157,10 +157,14 @@ func (e *PointGetExecutor) Next(ctx context.Context, req *chunk.Chunk) error {
 		tblID = e.tblInfo.ID
 	}
 	if e.idxInfo != nil {
-		e.idxKey, err = encodeIndexKey(e.base(), e.tblInfo, e.idxInfo, e.idxVals, tblID)
+		hasNull := false
+		e.idxKey, hasNull, err = encodeIndexKey(e.base(), e.tblInfo, e.idxInfo, e.idxVals, tblID)
 		if err != nil && !kv.ErrNotExist.Equal(err) {
 			return err
 		}
+		if hasNull {
+			return nil
```

Comment:
`x in (1,2,3,null)` return nil?

Reply:
This case will use `BatchGet` instad of `PointGet`:
```
mysql> select * from t where f in (null, 2);
+----+------+
| id | f    |
+----+------+
|  2 | 2    |
+----+------+
1 row in set (0.00 sec)

mysql> explain select * from t where f in (null, 2);
+-------------------+---------+------+-------------------------+------------------------------+
| id                | estRows | task | access object           | operator info                |
+-------------------+---------+------+-------------------------+------------------------------+
| Batch_Point_Get_1 | 2.00    | root | table:t, index:idx_f(f) | keep order:false, desc:false |
+-------------------+---------+------+-------------------------+------------------------------+
1 row in set (0.00 sec)

mysql> select * from t;
+----+------+
| id | f    |
+----+------+
|  1 | NULL |
|  2 | 2    |
+----+------+
2 rows in set (0.00 sec)
```